volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,646 @@
+            values = {
+                str(value.get_name()): value
+                for value in program_key.get_values()
+                if value.get_name() in [key.value for key in Win8ProgramValName]
+            }
+            vollog.debug(f"Parsing Win8 Program key {program_key.get_name()}")
+            program_id = program_key.get_name().strip().strip("\u0000")
+
+            product = _get_string_value(values, Win8ProgramValName.Product.value)
+            company = _get_string_value(values, Win8ProgramValName.Publisher.value)
+            install_time = _get_datetime_utc_epoch_value(
+                values, Win8ProgramValName.InstallTime.value
+            )
+            _version = _get_string_value(values, Win8ProgramValName.Version.value)
+
+            if isinstance(_version, str):
+                if isinstance(product, str):
+                    product = f"{product} {_version}"
+                else:
+                    product = f"UNKNOWN {_version}"
+
+            yield program_id, _AmcacheEntry(
+                AmcacheEntryType.Program.name,
+                company=company,
+                last_modify_time=conversion.wintime_to_datetime(
+                    program_key.LastWriteTime.QuadPart
+                ),
+                install_time=install_time,
+                product_name=product,
+            )
+
+    @classmethod
+    def parse_inventory_app_key(
+        cls, inv_app_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[str, _AmcacheEntry]]:
+        """Parses InventoryApplication entries from the Windows 10 `Root\\InventoryApplication` key.
+
+        :param programs_key: The `Root\\InventoryApplication` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\InventoryApplicationFile` entries, and the second member is the `AmcacheEntry`.
+        """
+        for program_key in inv_app_key.get_subkeys():
+            program_id = program_key.get_name()
+
+            values = {
+                str(value.get_name()): value
+                for value in program_key.get_values()
+                if value.get_name() in [key.value for key in Win10InvAppValName]
+            }
+
+            name = _get_string_value(values, Win10InvAppValName.Name.value)
+            version = _get_string_value(values, Win10InvAppValName.Version.value)
+            publisher = _get_string_value(values, Win10InvAppValName.Publisher.value)
+            path = _get_string_value(values, Win10InvAppValName.RootDirPath.value)
+            install_date = _get_datetime_str_value(
+                values, Win10InvAppValName.InstallDate.value
+            )
+            last_mod = conversion.wintime_to_datetime(
+                program_key.LastWriteTime.QuadPart
+            )
+
+            product: str = name if isinstance(name, str) else "UNKNOWN"  # type: ignore
+            if isinstance(version, str):
+                product += " " + version
```

Comment:
Should we not be returning `version` as a separate field, so another plugin could parse it easily?  I guess they're both strings, it just feels like if they're separate now, why combine them rather than pass them through as two fields?